src/main/java/io/vertx/core/http/impl/HttpClientResponseImpl.java

```diff
@@ -69,7 +69,7 @@
   HttpClientResponseImpl(Vertx vertx, HttpClientRequestImpl request, ClientConnection conn, HttpResponse response) {
     this.vertx = vertx;
     this.statusCode = response.getStatus().code();
-    this.statusMessage = response.getStatus().reasonPhrase();
+    this.statusMessage = response.getStatus().reasonPhrase().replaceAll("\\\\r", "\r").replaceAll("\\\\n", "\n");
```

Comment:
I'd caution against this, as it will cause replaceAll to executed twice for each client response which could kill performance.